addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlHandler.java

```diff
@@ -147,32 +191,165 @@ private void handleBrightnessCommand(NhcAction nhcAction, Command command) {
             }
         } else if (command instanceof PercentType) {
             PercentType p = (PercentType) command;
-            nhcAction.execute(p.intValue());
+            if (p == PercentType.ZERO) {
+                nhcAction.execute(NHCOFF);
+            } else {
+                nhcAction.execute(p.intValue());
+            }
         }
     }
 
     private void handleRollershutterCommand(NhcAction nhcAction, Command command) {
-        if (command instanceof UpDownType) {
-            UpDownType s = (UpDownType) command;
-            if (s == UpDownType.UP) {
-                nhcAction.execute(255);
-            } else {
-                nhcAction.execute(254);
-            }
-        } else if (command instanceof StopMoveType) {
-            StopMoveType s = (StopMoveType) command;
-            if (s == StopMoveType.STOP) {
-                nhcAction.execute(253);
-            }
-        } else if (command instanceof PercentType) {
-            PercentType p = (PercentType) command;
-            int currentState = nhcAction.getState();
-            if (currentState < p.intValue()) {
-                nhcAction.execute(255);
-            } else if (currentState > p.intValue()) {
-                nhcAction.execute(254);
+        Configuration config = this.getConfig();
+        boolean invert = (boolean) config.get(CONFIG_INVERT);
+        logger.trace("handleRollerShutterCommand: rollershutter {} command {}", config.get(CONFIG_ACTION_ID), command);
```

Comment:
Note that you always perform those gets, please wrap the three statements in a isTraceEnabled